src/ethereumProofPointIssuer.ts

```diff
@@ -0,0 +1,275 @@
+   * @param [validFromDate] Optional date from which the issued Proof Point will be valid. If null then there is no earliest date at which the Proof Point is valid.
+   * @param [validUntilDate] Optional date until which the issued Proof Point will be valid. If null then there is no latest date at which the Proof Point is valid.
+   * @returns A @ProofPointIssueResult describing the result of the action.
+   */
+  async issue(
+    type: string,
+    issuer: string,
+    content: unknown,
+    validFromDate: Date | null = null,
+    validUntilDate: Date | null = null
+  ): Promise<ProofPointIssueResult> {
+    return this._issue(
+      type,
+      issuer,
+      content,
+      (
+        c: EthereumProofPointRegistry,
+        id: ProofPointId,
+        issuer: EthereumAddress
+      ) => c.issue(id, issuer),
+      validFromDate,
+      validUntilDate
+    );
+  }
+
+  /**
+   * Commit a new Proof Point
+   * @param type A URI string identifying the type of Proof Point to issue. This may be one of the values defined in the Provenance ontology.
+   * @param issuer A string identifying the Ethereum address from which to commit the Proof Point. This may be either an Ethereum address or a did:web URI. It must represent an account that you control and must be sufficiently funded to pay for the issuance transaction.
+   * @param content A javascript object representing the type specific content of the payload. The shape of the data should conform to the specification of the @param type parameter.
+   * @param [validFromDate] Optional date from which the issued Proof Point will be valid. If null then there is no earliest date at which the Proof Point is valid.
+   * @param [validUntilDate] Optional date until which the issued Proof Point will be valid. If null then there is no latest date at which the Proof Point is valid.
+   * @returns A @ProofPointIssueResult describing the result of the action.
+   */
+  async commit(
+    type: string,
+    issuer: string,
+    content: unknown,
+    validFromDate: Date | null = null,
+    validUntilDate: Date | null = null
+  ): Promise<ProofPointIssueResult> {
+    return this._issue(
+      type,
+      issuer,
+      content,
+      (
+        c: EthereumProofPointRegistry,
+        id: ProofPointId,
+        issuer: EthereumAddress
+      ) => c.commit(id, issuer),
+      validFromDate,
+      validUntilDate
+    );
+  }
+
+  /**
+   * Revoke a Proof Point identified by it's ID. You must control the account that originally issued the Proof Point. The account must be sufficiently funded to execute the revoke transaction.
+   * @param proofPointId The ID of the Proof Point to revoke. This is the value returned in the @param proofPointId field of the {@link ProofPointIssueResult} when the Proof Point was issued.
+   */
+  async revoke(proofPointId: ProofPointId): Promise<void> {
```

Comment:
Seems odd that an Issuer can revoke (re API point above).

Reply:
I see what you mean. Can you suggest a better name for the class? In one sense it kind of makes sense as in the entity that is the Issuer is the only entity that can revoke a given PP.